### DISCUSSION TITLE
Brackets Appear Red in Generated Assembly File (.asm)

### DISCUSSION THREAD

**Discussion by DS-567** [stnolting/neorv32#534] [General]
Hi everyone,

I have been writing various C programs for NEORV32 and comparing it the compiled and assembled binaries shown in the main.asm file. But I have came across in various places where some of the brackets (both curly and normal) appear red and others do not. This may be a general assembly question more so than a RISC-V question, but figured I would ask it here because I am using NEORV32.

I noticed this because I had a few functions that were called from main(), but as part of an experiment I added the function code body inside main() so there was no function call to executer the same logic. When I looked at the assembly it appeared like this:

![image](https://user-images.githubusercontent.com/100368901/223172057-ec9423e6-9f01-45f8-9beb-c8e019df60a2.png)

The code compiled fine but maybe this is an optimization by the compiler to remove the code. I checked the demo_blink_led assembled asm file and there appears a red bracket here too (see below). 

Is there a specific reason they appear red? Or could it be to do with not having the correct RISC-V extensions enabled?

![image](https://user-images.githubusercontent.com/100368901/223168800-32d90d9c-18b1-4134-a243-55c14bf18641.png)

**Comment by stnolting** [comment]
I have just checked an assembly file generated by the NEORV32 software framework and there is no "formatting" data encoded within the file - only raw ASCII (UTF-8) stuff. So maybe this is just the syntax highlighting of your editor? 😅

**Reply by DS-567** [reply]
Ah, you are correct. I opened using a different editor and the parentheses are no longer strange. Must be a syntax issue with VSC 🤨. Thanks Stephan.
